subsystems/networking/networkmanager_linux.go

```diff
@@ -861,21 +872,29 @@ func (n *Networking) mainLoop(ctx context.Context) {
 				if n.Config().TurnOnHotspotIfWifiHasNoInternet.Get() {
 					lastConnectivity = n.connState.getLastOnline()
 				}
+			} else {
+				// Did not get what we need in previous provision (e.g. bad wifi password).
+				forceStartPMode = true
+				// Stay in pMode until user input.
+				// We get to this point with things like incorrect ssid or password that will likely not auto-resolve.
+				// in bad password case: haveCandidates condition for exiting pMode remains true (likely stale).
+				forceStayInPModeUntilRecvUserInput = true
 			}
 		}
 
-		shouldReboot := n.Config().DeviceRebootAfterOfflineMinutes > 0 &&
+		shouldRebootSystem := n.Config().DeviceRebootAfterOfflineMinutes > 0 &&
 			lastConnectivity.Before(now.Add(time.Duration(n.Config().DeviceRebootAfterOfflineMinutes)*-1))
 
-		if shouldReboot && n.doReboot(ctx) {
+		if shouldRebootSystem && n.doReboot(ctx) {
 			return
 		}
 
 		hitOfflineTimeout := now.After(lastConnectivity.Add(time.Duration(n.Config().OfflineBeforeStartingHotspotMinutes))) &&
 			now.After(pModeChange.Add(time.Duration(n.Config().OfflineBeforeStartingHotspotMinutes)))
 		// not in provisioning mode, so start it if not configured (/etc/viam.json)
 		// OR as long as we've been offline AND out of provisioning mode for at least OfflineTimeout (2 minute default)
-		if !isConfigured || hitOfflineTimeout {
+		if !isConfigured || hitOfflineTimeout || forceStartPMode {
+			forceStartPMode = false
```

Comment:
```suggestion
		if !isConfigured || hitOfflineTimeout || forceStartPMode {
```
Tiny nit, but the boolean is declared inside the loop, so it already reset to false each iteration when it's redeclared.